packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts

```diff
@@ -553,6 +598,30 @@ export class FileSystem extends FileSystemBase {
     if (props.throughputMode === ThroughputMode.ELASTIC && props.performanceMode === PerformanceMode.MAX_IO) {
       throw new Error('ThroughputMode ELASTIC is not supported for file systems with performanceMode MAX_IO');
     }
+
+    if (props.replicationConfiguration) {
+      const { destinationFileSystem, region, availabilityZone, kmsKey } = props.replicationConfiguration;
+      if (props.replicationOverwriteProtection === ReplicationOverwriteProtection.DISABLED) {
```

Comment:
do we need to add validation to check that if `destinationFileSystem` is being passed in then it should be read-only (i.e.  replicationOverwriteProtection set to `DISABLED`)?

Reply:
I understand your point, but the IFileSystem interface does not include a replicationOverwriteProtection property. Could we instead use the FileSystem class for the destinationFileSystem parameter?
(I generally think it's common to receive an ISomeClass as an argument.)